### PR TITLE
CicleCI integration

### PR DESCRIPTION
**Pull request by ziafazal**
This PR has changes to integrate CircleCI version 2.0
It performs three checks for each PR

1. Runs pyton unit test for `edly_panel_app` and `openedx.features.edly`
2. Runs Pylint checks for branch diff
3. Runs Pep8 checks for branch diff

We have to skip recommender-xblock in requirements/edx/testing.txt because it has an issue which does not allow it to be installed on CircleCI. This xblock is not being used by edly.

Error trace of recommender-xblock
```
   writing manifest file 'recommender_xblock.egg-info/SOURCES.txt'
    Copying recommender_xblock.egg-info to /home/circleci/circleci-edx-platform/venv/lib/python2.7/site-packages/recommender_xblock-1.4.0-py2.7.egg-info
    running install_scripts
    writing list of installed files to '/tmp/pip-record-FgYmJZ/install-record.txt'
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-zcHs1Y/recommender-xblock/setup.py", line 80, in <module>
        "Programming Language :: Python :: 2.7",
      File "/home/circleci/circleci-edx-platform/venv/lib/python2.7/site-packages/setuptools/__init__.py", line 145, in setup
        return distutils.core.setup(**attrs)
      File "/usr/local/lib/python2.7/distutils/core.py", line 151, in setup
        dist.run_commands()
      File "/usr/local/lib/python2.7/distutils/dist.py", line 953, in run_commands
        self.run_command(cmd)
      File "/usr/local/lib/python2.7/distutils/dist.py", line 972, in run_command
        cmd_obj.run()
      File "/tmp/pip-install-zcHs1Y/recommender-xblock/setup.py", line 17, in run
        self.compile_translations()
      File "/tmp/pip-install-zcHs1Y/recommender-xblock/setup.py", line 33, in compile_translations
        self.announce('Translations compilation failed: %s', ex.message)
      File "/usr/local/lib/python2.7/distutils/cmd.py", line 184, in announce
        log.log(level, msg)
      File "/usr/local/lib/python2.7/distutils/log.py", line 34, in log
        self._log(level, msg, args)
      File "/usr/local/lib/python2.7/distutils/log.py", line 21, in _log
        raise ValueError('%s wrong log level' % str(level))
    ValueError:  wrong log level
error
ERROR: Command errored out with exit status 1: /home/circleci/circleci-edx-platform/venv/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-zcHs1Y/recommender-xblock/setup.py'"'"'; __file__='"'"'/tmp/pip-install-zcHs1Y/recommender-xblock/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' install --record /tmp/pip-record-FgYmJZ/install-record.txt --single-version-externally-managed --compile --install-headers /home/circleci/circleci-edx-platform/venv/include/site/python2.7/recommender-xblock Check the logs for full command output.
Exception information:
```

### CODE REVIEW

File: openedx/features/edly/tests/test_middleware.py [hunk 1]
```diff
@@ -27,6 +27,14 @@ def setUp(self):
         self.request = RequestFactory().get('/')
         self.request.user = self.user
         self.request.site = SiteFactory()
+        self.site_config = SiteConfigurationFactory(
+            site=self.request.site,
+            enabled=True,
```
Comment (rehan99000) [inline openedx/features/edly/tests/test_middleware.py:32]: we don't need to do this. `SiteConfigurationFactory` handles it by default.

File: requirements/edx/development.txt [hunk 2]
```diff
@@ -42,7 +42,7 @@ git+https://github.com/edx-solutions/xblock-drag-and-drop-v2@v2.1.6#egg=xblock-d
 -e git+https://github.com/edx-solutions/xblock-google-drive.git@138e6fa0bf3a2013e904a085b9fed77dab7f3f21#egg=xblock-google-drive
 git+https://github.com/open-craft/xblock-poll@add89e14558c30f3c8dc7431e5cd6536fff6d941#egg=xblock-poll==1.5.1
 -e git+https://github.com/raccoongang/xblock-pdf.git@sgab-v.1.0.1#egg=xblock-pdf
-git+ssh://git@github.com/edly-io/edly-panel-edx-app.git@develop-multisite#egg=edly-panel-app
+git+ssh://git@github.com/edly-io/edly-panel-edx-app.git@3.0.4#egg=edly-panel-app==3.0.4
```
Comment (rehan99000) [inline requirements/edx/development.txt:45]: why do we need this change?

Comment (ziafazal) [reply requirements/edx/development.txt:45]: It is already being used in `development.txt` already. I think it is used when devstack is setup.

File: openedx/features/edly/tests/test_middleware.py (hunk 1 above)
Comment (ziafazal) [reply openedx/features/edly/tests/test_middleware.py:32]: let me remove `enabled=True`

File: requirements/edx/development.txt (hunk 2 above)
Comment (rehan99000) [reply requirements/edx/development.txt:45]: Yes. but we need it to install the `develop-multisite` branch for this repo, not a release tag

Comment (ziafazal) [reply requirements/edx/development.txt:45]: Actaully I'm utilizing requirements cache feature of circleci. If we use `develop-multisite` branch which means we don't have to change `edx/requirements/testing.txt` and as result of it circleci would use cached requirements which can break units tests.

**Pull request merged**